File: cartipay-main/Pardakht.PardakhtPay/BackEnd/Pardakht.PardakhtPay.Shared.Models/Entities/BankStatementItem.cs
using System;
using System.ComponentModel.DataAnnotations;
using Pardakht.PardakhtPay.Shared.Models.Models;

namespace Pardakht.PardakhtPay.Shared.Models.Entities
{
    public class BankStatementItem : BaseEntity
    {
        public int RecordId { get; set; }

        public DateTime CreationDate { get; set; }

        public int AccountId { get; set; }

        [Required]
        public string AccountGuid { get; set; }

        public int LoginId { get; set; }

        [Required]
        [StringLength(50)]
        public string LoginGuid { get; set; }

        [StringLength(50)]
        public string TransactionNo { get; set; }

        [StringLength(50)]
        public string CheckNo { get; set; }

        public DateTime TransactionDateTime { get; set; }

        public decimal Debit { get; set; }

        public decimal Credit { get; set; }

        public decimal Balance { get; set; }

        [StringLength(500)]
        public string Description { get; set; }

        public DateTime InsertDateTime { get; set; }

        public int ConfirmedTransactionId { get; set; }

        [StringLength(100)]
        public string Notes { get; set; }

        public int? WithdrawalId { get; set; }

        //[ForeignKey(nameof(ConfirmedTransactionId))]
        //public Transaction Transaction { get; set; }
    }
}

File: cartipay-main/Pardakht.PardakhtPay/FrontEnd/Pardakht.PardakhtPay.AdminPortal/src/app/models/bank-statement-item.ts
import { ListSearchArgs, AgGridSearchArgs } from "./types";

export class BankStatementItem {
    id: number;
    creationDate: Date;
    accountId: number;
    accountGuid: string;
    transactionNo: string;
    checkNo: string;
    transactionDateTimeStr: string;
    debit: number;
    credit: number;
    balance: number;
    description: string;
    insertDateTime: Date;
    usedDateStr: string;
    transactionId: number;
    isRisky: boolean;
    withdrawalId: number;
}

export class BankStatementItemSearchArgs extends AgGridSearchArgs {
    filterModel: any;
    statementItemType: number;
    accountGuids: string[];
    isRisky: boolean;
}